exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -128,55 +148,184 @@ void createTestKitWithoutServices() {
 
-    @Override
-    public short getId() {
-      return SERVICE_ID;
-    }
+    TransactionMessage message = constructTestTransactionMessage("Test message");
+    Block block = testKit.createBlockWithTransaction(message);
+    assertThat(block.getNumTransactions()).isEqualTo(1);
 
-    @Override
-    public String getName() {
-      return SERVICE_NAME;
-    }
+    // Two blocks were created, so two afterCommit transactions should be submitted into pool
+    List<TransactionMessage> inPoolTransactions = testKit
+        .findTransactionsInPool(tx -> tx.getServiceId() == TestService.SERVICE_ID);
+    assertThat(inPoolTransactions).hasSize(2);
+  }
 
-    @Override
-    public Transaction convertToTransaction(RawTransaction rawTransaction) {
-      throw new UnsupportedOperationException();
-    }
+  @Test
+  void createBlockWithTransaction() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+
+    TransactionMessage message = constructTestTransactionMessage("Test message");
+    Block block = testKit.createBlockWithTransaction(message);
+    assertThat(block.getNumTransactions()).isEqualTo(1);
+
+    testKit.withSnapshot((view) -> {
+      Blockchain blockchain = Blockchain.newInstance(view);
+      assertThat(blockchain.getHeight()).isEqualTo(1);
+      assertThat(block).isEqualTo(blockchain.getBlock(1));
+      Map<HashCode, TransactionResult> transactionResults = toMap(blockchain.getTxResults());
+      assertThat(transactionResults).hasSize(1);
+      TransactionResult transactionResult = transactionResults.get(message.hash());
+      assertThat(transactionResult).isEqualTo(TransactionResult.successful());
+      return null;
+    });
+  }
+
+  @Test
+  void createBlockWithTransactions() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+
+    TransactionMessage message = constructTestTransactionMessage("Test message");
+    TransactionMessage message2 = constructTestTransactionMessage("Test message 2");
+
+    Block block = testKit.createBlockWithTransactions(ImmutableList.of(message, message2));
+    assertThat(block.getNumTransactions()).isEqualTo(2);
+
+    testKit.withSnapshot((view) -> {
+      Blockchain blockchain = Blockchain.newInstance(view);
+      assertThat(blockchain.getHeight()).isEqualTo(1);
+      assertThat(block).isEqualTo(blockchain.getBlock(1));
+      Map<HashCode, TransactionResult> transactionResults = toMap(blockchain.getTxResults());
+      assertThat(transactionResults).hasSize(2);
+
+      TransactionResult transactionResult = transactionResults.get(message.hash());
+      assertThat(transactionResult).isEqualTo(TransactionResult.successful());
+      TransactionResult transactionResult2 = transactionResults.get(message2.hash());
+      assertThat(transactionResult2).isEqualTo(TransactionResult.successful());
+      return null;
+    });
+  }
+
+  @Test
+  void createBlockWithTransactionsVarargs() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
```

Comment:
The code below seems to be the same as above except the function. I'd consider de-duplicating.